vertica/datadog_checks/vertica/vertica.py

```diff
@@ -626,3 +622,44 @@ def iter_rows(self, view):
 
         for row in cursor.iterate():
             yield row
+
+    def parse_metric_groups(self):
+        # If you create a new function, please add this to `default_metric_groups` below and
+        # the config file (under `metric_groups`).
+        default_metric_groups = OrderedDict(
+            (
+                ('licenses', self.query_licenses),
+                ('license_audits', self.query_license_audits),
+                ('system', self.query_system),
+                ('nodes', self.query_nodes),
+                ('projections', self.query_projections),
+                ('projection_storage', self.query_projection_storage),
+                ('storage_containers', self.query_storage_containers),
+                ('host_resources', self.query_host_resources),
+                ('query_metrics', self.query_query_metrics),
+                ('resource_pool_status', self.query_resource_pool_status),
+                ('disk_storage', self.query_disk_storage),
+                ('resource_usage', self.query_resource_usage),
+            )
+        )
+
+        metric_groups = self.instance.get('metric_groups') or list(default_metric_groups)
+
+        # Ensure all metric groups are valid
+        invalid_groups = []
+
+        for group in metric_groups:
+            if group not in default_metric_groups:
+                invalid_groups.append(group)
+
+        if invalid_groups:
+            raise ConfigurationError(
+                'Invalid metric_groups found in vertica conf.yaml: {}'.format(', '.join(invalid_groups))
+            )
+
+        # License query needs to be run before getting system
+        if 'system' in metric_groups and 'licenses' not in metric_groups:
+            self.log.debug('Detected `system` metric group, adding the `licenses` to metric_groups.')
+            metric_groups.insert(0, 'licenses')
+
+        self._metric_groups.extend(default_metric_groups[group] for group in metric_groups)
```

Comment:
If a user specify groups manually, we still want them to be collected in the right order. Otherwise the check might query `licenses` after `system`.

I think this suggestion should solve the issue

```suggestion
        self._metric_groups.extend(default_metric_groups[group] for group in default_metric_groups if group in metric_groups)
```